backend/bittan/bittan/views/views.py

```diff
@@ -1,59 +1,204 @@
-
-
-    
-    
+from bittan.models.payment import PaymentStatus
+from bittan.services.swish.swish_payment_request import SwishPaymentRequest, PaymentStatus as SwishPaymentStatus
+
+from ..models import ChapterEvent, Ticket, TicketType, Payment
+
+from bittan.services.swish.swish import Swish
+
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.request import Request
+from rest_framework.decorators import api_view
+from rest_framework import serializers
+
+import random
+
+from django.db import models
+from django.utils import timezone
+from django.db.utils import IntegrityError
+from django.db.models import Sum
+from django.core.exceptions import ObjectDoesNotExist
+
+import logging
+
+class TicketsSerializer(serializers.Serializer):
+    ticket_type = serializers.CharField(required=True)
+    count = serializers.IntegerField(required=True, min_value=0)
+
+class ReserveTicketRequestSerializer(serializers.Serializer):
+    chapter_event = serializers.CharField(required = True)
+    tickets = serializers.ListField(child=TicketsSerializer())
+
+
+@api_view(['POST'])
+def reserve_ticket(request: Request) -> Response:
+    response_data: dict
+
+    valid_ser = ReserveTicketRequestSerializer(data=request.data)
+    if valid_ser.is_valid():
+        response_data = valid_ser.validated_data
+    else:
+        return Response(
+                "InvalidRequestData",
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+    event_id: int = response_data["chapter_event"]
+    tickets: list = response_data["tickets"]
+    
+    reservation_count: int = sum(x["count"] for x in tickets)
+    
+    try:
+        chapter_event: ChapterEvent = ChapterEvent.objects.get(id=event_id)
+    except (ObjectDoesNotExist, ValueError):
+        return Response(
+            "EventDoesNotExist",
+            status=status.HTTP_404_NOT_FOUND
+        )
+    
+    if reservation_count > chapter_event.max_tickets - chapter_event.alive_ticket_count:
+        return Response(
+            "OutOfTickets", 
+            status=status.HTTP_403_FORBIDDEN
+        )
+
+    payment = Payment.objects.create(
+        expires_at = timezone.now() + chapter_event.reservation_duration,
+        swish_id = None, 
+        status = PaymentStatus.RESERVED,
+        email = None, 
+    )
+
+    for ticket in tickets:
+        for _ in range(ticket["count"]):
+            # Attempts to create a ticket a maximum of 1000 times. This is to ensure that the external_id of the ticket is guaranteed to be unique. 
+            for _ in range(1000):
+                try:
+                    Ticket.objects.create(
+                            external_id=''.join(random.choice("ABCDEFGHIJKLMNOPQRSTUVWXYZ") for _ in range(6)),
+                            time_created=timezone.now(),
+                            payment=payment,
+                            ticket_type=TicketType.objects.get(title=ticket["ticket_type"], is_visible=True)
```

Comment:
Yes, I have thought about this. I think I will change the view to take the `pk` of the ticket instead of the `ticket_type`. I think this would be a better solution and can currently see no downsides.